File: src/main/java/ReverseLinkedList/BIGBOSS/ReverseLinkedList.cpp
//
// Created by BIGBOSS on 2019-06-27.
//

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution
{
    public:

    ListNode* reverseList(ListNode* head)
    {
        ListNode* last = nullptr;

        while (head)
        {
            ListNode* tmp = head;
            head = head->next;
            tmp->next = last;
            last = tmp;
        }

        return last;
    }
};

File: src/main/java/ReverseLinkedList/lion/ReverseLinkedList.java
package ReverseLinkedList.lion;

public class ReverseLinkedList {
	public ListNode reverseList(ListNode head) {
		ListNode p = null;
		ListNode q = head;
		while (head != null) {
			head = head.next;
			q.next = p;
			p = q;
			q = head;
		}
		return p;
	}
}

class ListNode {
	int val;
	ListNode next;

	ListNode(int x) {
		val = x;
	}
}